app/uk/gov/hmrc/apisubscriptionfields/service/FieldsDefinitionService.scala

```diff
@@ -41,11 +41,14 @@ class FieldsDefinitionService @Inject() (repository: FieldsDefinitionRepository)
     } yield fetch.map(asResponse)
   }
 
-  def getAll: Future[List[FieldsDefinitionResponse]] = {
+  def getAll: Future[FieldsDefinitionResponse] = {
     Logger.debug(s"[getAll]")
-    for {
+    (for {
       list <- repository.fetchAll()
-    } yield list.map(asResponse)
+    } yield list map
```

Comment:
Too many lines of code
I am sure you can write it in a more concise way

Reply:
I have simplified to 
``` scala
    (for {
      list <- repository.fetchAll()
    } yield list map (_.fieldDefinitions)) map (_.flatten) map (FieldsDefinitionResponse(_))
```